### PR TITLE
Support numeric (and other token) names and ids

### PR DESCRIPTION
**Pull request by cmoesel**
Replace `SEQUENCE` in the parser grammar with `name`, which allows `SEQUENCE` plus a number of other tokens that authors might choose to use as a name or id.

The FHIR-based rules about names are still followed in the importer, but note that they've been demoted to warnings since FHIR actually uses a "SHOULD" not a "MUST" in its language about the name regex.

This also fixes a couple of null-dereference errors I ran into during testing (in the error listener and importer).

I'd like for @mint-thompson to be one of the reviewers since they are very familiar w/ the grammar and the types of problems we are trying to solve here.

### CODE REVIEW

**Review by jafeltra — COMMENTED** [review]
This looks good to me! @mint-thompson definitely has deeper knowledge of the issues and grammar, so I agree they would be a good reviewer here!

The one minor thing these changes made me realize is that we don't log that warning about the name regular expression when we create Instances, Mappings, Invariants, and RuleSets. I think this might just be because we don't have the same process, and maybe not even the same regular expression applies. But I wanted to double check.

**Comment by cmoesel** [comment]
> The one minor thing these changes made me realize is that we don't log that warning about the name regular expression when we create Instances, Mappings, Invariants, and RuleSets.

In Instances and Invariants, the name you put after the `:` in the declaration maps to an `id` type -- so it has different rules.  That said, `id` is actually stricter than name because it only allows 64 characters, so perhaps we should be warning on those (if we're not).

In Mappings and RuleSets, the name you put after the `:` is just a handle in FSH; it does not actually map to any specific field in FHIR.

**Review by jafeltra — DISMISSED** [review]
Ah, okay, that all makes sense to me! I'm approving! I haven't checked to see if there was a warning for the character limit, but that's not this task anyway!

**Review by mint-thompson — CHANGES_REQUESTED** [review]
I like what you did with the `name` parser token. Could you add some more tests for some of the other lexer tokens that `name` can match?

File: src/import/parserContexts.ts [hunk 1]
```diff
@@ -208,6 +208,10 @@ export interface SdRuleContext extends ParserRuleContext {
   insertRule(): InsertRuleContext;
 }
 
+// NameContext can be so many things, but we really only care about its text,
+// so just supporting getText() should be sifficient (thus ParserRuleContext)
```
[the remark this answers is not in the document]
Comment (cmoesel) [reply src/import/parserContexts.ts:212]: Oops.

**Comment by cmoesel** [comment]
> I like what you did with the `name` parser token. Could you add some more tests for some of the other lexer tokens that `name` can match?

Sure, but I don't want to get too silly, so I'll probably only do it in one of the places `name` gets used rather than all of them.  Does that seem reasonable?

**Comment by mint-thompson** [comment]
@cmoesel that sounds just right, since then together it would show where and how name can be used without having to exhaust the complete set of scenarios.

**Comment by cmoesel** [comment]
Thanks for making me write tests for the other tokens, @mint-thompson.  As it turns out, I had used the `KW_VALUESET` token when I needed to use the `KW_VSREFERENCE` token.  I would not have caught that but for writing the tests!  Good call!

BTW -- should be ready for re-review @mint-thompson and @jafeltra.

**Review by mint-thompson — APPROVED** [review]

**Review by jafeltra — APPROVED** [review]

**Pull request merged**